backend/api/Services/RobotService.cs

```diff
@@ -30,7 +30,7 @@ public RobotService(FlotillaDbContext context)
 
         private IQueryable<Robot> GetRobotsWithSubModels()
         {
-            return _context.Robots.Include(r => r.VideoStreams).Include(r => r.Model);
+            return _context.Robots.Include(r => r.VideoStreams).Include(r => r.Model).Include(r => r.CurrentAssetDeck);
```

Comment:
should this be added in flotilladbcontext?

Reply:
In my understanding this is the place where we are including sub models in the context object and inside FlotilladB we add constraints on the objects of these sub models. Atleast this is what I think happens with the Class Assetdeck. I would keep these two things seperate for clarity.

Reply:
I think context definition should live in the db context - currentassetdeck is always a property that lives on the robot model if I have not misunderstood

Reply:
Anyways refactoring of this can be another issue.